django/thunderstore/repository/views/package/list.py

```diff
@@ -250,6 +254,13 @@ def get_queryset(self):
             )
         )
 
+        # If the URL has a query parameter for blob (/?blob=X), filter out all packages that don't include that blob
+        blob_query_parameter = self.get_blob_query()
+        if blob_query_parameter:
+            queryset = queryset.filter(
+                package__latest__file_tree__entries__blob__checksum_sha256=blob_query_parameter
+            ).distinct()
```

Comment:
Use `.exclude` rather than `.filter`, reason being that chaining excludes works more along the lines you'd expect whereas chaining filters has a few unintuitive behaviors you'll probably fail to account for (and in fact might already cause issues here, I wouldn't be able to say without spending too much time thinking about it).

So in short, chaining excludes is more foolproof & easier to review